src/main/java/me/karakelley/http/controllers/Application.java

```diff
@@ -3,26 +3,36 @@
 import me.karakelley.http.PublicDirectory;
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
+import me.karakelley.http.utility.HtmlGenerator;
 
 public class Application implements Controller {
   private final Router router = new Router();
+  private PublicDirectory publicDirectory = null;
 
   public Application() {
     router.route("/", new HelloWorldController());
     setupCommonRoutes();
   }
 
   public Application(PublicDirectory publicDirectory) {
-    router.route("/", new StaticFilesController());
+    this.publicDirectory = publicDirectory;
+    router.route("/", new StaticFilesController(publicDirectory, new HtmlGenerator()));
     setupCommonRoutes();
   }
 
   @Override
   public Response respond(Request request) {
+    checkRequestedPath(request);
```

Comment:
Check requested path for what? Why are we checking it?

Reply:
I need to check if the requested resource exists

Reply:
Ok. Cool. I'd say a better function name would be `checkIfRequestedResourceExists`. That name encapsulates the "why" but it still doesn't really explain "ok, then what." You can generally say that something will return a `boolean` by phrasing it like a predicate `requestedResourceExists` and you can convey that a message will throw by prepending it with `ensure` or `assert`. `ensureTheRequestedResourceExists` or `assertTheRequestedResourceExists`. In this case I think you're just adding a route to the router, so I'd probably rename this altogether and go with `addRouteForRequestedResource`.